server/webapp/WEB-INF/rails.new/app/controllers/api_v4/notification_filters_controller.rb

```diff
@@ -0,0 +1,78 @@
+##########################################################################
+# Copyright 2017 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+java_import "com.thoughtworks.go.domain.NotificationFilter",
+            "com.thoughtworks.go.domain.StageEvent",
+            "com.thoughtworks.go.domain.exception.UncheckedValidationException"
+
+module ApiV4
+  class NotificationFiltersController < ApiV4::BaseController
+    before_action :check_user_and_404
+    before_action :load_user
+    before_action :check_filter_params, only: :create
+
+    def index
+      render json: filters_for_current_user.to_json
+    end
+
+    def create
+      filter = filter_from_params
+      user_service.add_notification_filter(@user.id, filter)
+      @user = user_service.load(@user.id)
+      render json: filters_for_current_user.to_json
```

Comment:
All responses from all controllers are expected to have a `Content-Type: application/vnd.go.cd.#{version}+json` header, `render json` would be considered unacceptable from an API perspective.

 
`render DEFAULT_FORMAT => filters_for_current_user.to_json` should do all the magic for you.